probe-rs/src/debug/registers.rs

```diff
@@ -234,14 +296,28 @@ impl DebugRegisters {
             .unwrap_or_else(|| "unknown register".to_string())
     }
 
-    /// Retrieve a register by searching against either the name or the special_name.
-    pub fn get_register_by_name(&self, register_name: &str) -> Option<DebugRegister> {
-        self.0
-            .iter()
-            .find(|&debug_register| {
-                debug_register.name == register_name
-                    || debug_register.special_name == Some(register_name)
-            })
-            .cloned()
+    /// Retrieve a register by searching against either the name or the alias.
+    pub fn get_register_by_name(&self, register_name: &str) -> Option<&DebugRegister> {
+        self.0.iter().find(|&debug_register| {
+            debug_register.name == register_name || debug_register.alias == Some(register_name)
+        })
+    }
+
+    /// Update the `RegisterValue` of a register, identified by searching against either the name or the alias.
+    pub fn update_register_value_by_name(
```

Comment:
I would prefer to have a "typed" interface here, it shouldn't be necessary to refer to registers by strings, since we already know all the possible registers.